src/filters/truncate.js

```diff
@@ -2,30 +2,33 @@
 
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
-    return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+    return function truncate(str, charLimit, useWordBoundary, newlineLimit, removeExtraChars) {
```

Comment:
Yeah that one is a doozy, not sure what a good variable name is... could actually spell out ```characters``` but thinking that isn't going to help much.  Could aim for something like ```makeSpaceForTheShowHideLinkPlacement``` =)... Up for recommendations on it